tools/clang/unittests/HLSL/OptionsTest.cpp

```diff
@@ -380,3 +384,219 @@ TEST_F(OptionsTest, TestPreprocessOption) {
   VerifyPreprocessOption("/T ps_6_0 -P out.pp input.hlsl", "out.pp", Warning);
   VerifyPreprocessOption("/T ps_6_0 input.hlsl -P out.pp ", "out.pp", Warning);
 }
+
+static void VerifySerializeDxilFlags(llvm::StringRef command,
+                                     uint32_t ExpectFlags) {
+  std::string errorString;
+  const llvm::opt::OptTable *optionTable = getHlslOptTable();
+  llvm::SmallVector<llvm::StringRef, 4> args;
+  command.split(args, " ", /*MaxSplit*/ -1, /*KeepEmpty*/ false);
+  args.emplace_back("-Tlib_6_3");
+  args.emplace_back("input.hlsl");
+  MainArgs argStrings(args);
+  DxcOpts dxcOpts;
+  llvm::raw_string_ostream errorStream(errorString);
+
+  int retVal =
+      ReadDxcOpts(optionTable, DxcFlags, argStrings, dxcOpts, errorStream);
+  EXPECT_EQ(retVal, 0);
+  errorStream.flush();
+  printf("%s\n", errorString.c_str());
+  EXPECT_TRUE(errorString.empty());
+  EXPECT_EQ(
+      static_cast<uint32_t>(hlsl::options::ComputeSerializeDxilFlags(dxcOpts)),
+      ExpectFlags);
+}
+
+static uint32_t CombineFlags(llvm::ArrayRef<SerializeDxilFlags> flags) {
+  uint32_t result = 0;
+  for (SerializeDxilFlags f : flags)
+    result |= static_cast<uint32_t>(f);
+  return result;
+}
+
+struct SerializeDxilFlagsTest {
+  const char *command;
+  uint32_t flags;
+};
+
+using F = SerializeDxilFlags;
+
+TEST_F(OptionsTest, SerializeDxilFlags) {
+  // Test cases for SerializeDxilFlags
```

Comment:
I think it would be good to add a comment here about why these flag combinations were chosen. There is a lot of tests here, which is good, but its not clear how/why these ones in particular are chosen. I'm failing to see a comprehensive pattern just by looking at them.